### PR TITLE
Feature/type annotations

### PR DESCRIPTION
**Pull request by CFlaniganMide**
Pull request for python 3.6 style type annotations.  Leaving as a draft for now because it breaks compatibility with 3.5.

### CODE REVIEW

**Comment by StokesMIDE** [comment]
It looks like 25% of Python 3 users are still using versions earlier than 3.6; since type annotation is just unenforced syntactic sugar, I'd like to hold off until almost everyone is using 3.6 or newer. When the numbers improve, we can make the change - after documenting which ebmlite versions are compatible with which Python versions.

**Comment by CFlaniganMide** [comment]
Agreed, I'm going to leave this as a draft for now.  If someone really wants it, this is available.

File: ebmlite/core.py [hunk 1]
```diff
@@ -65,12 +66,12 @@
...
 # When `loadSchema()` is used, it will search these paths, in order, to find
 # the schema file.
-SCHEMA_PATH = ['',
+SCHEMA_PATH: List[str] = ['',
                os.path.realpath(os.path.dirname(schemata.__file__))]
 
 # SCHEMATA: A dictionary of loaded schemata, keyed by filename. Used by
 # `loadSchema()`. In most cases, SCHEMATA should not be otherwise modified.
-SCHEMATA = {}
+SCHEMATA: Dict[Union[str, bytes, bytearray], Any] = {}
```
Comment (StokesMIDE) [inline ebmlite/core.py:74]: I think the keys will always be string.

File: ebmlite/encoding.py [hunk 2]
```diff
@@ -22,28 +23,28 @@
...
-                   0x200000,
-                   0x10000000,
-                   0x0800000000,
-                   0x040000000000,
-                   0x02000000000000,
-                   0x0100000000000000
-                   ]
+DEFAULT_FLOAT_SIZE: int = 4 if sys.maxsize <= 2147483647 else 8
+
+LENGTH_PREFIXES: List[int] = [0,
```
Comment (StokesMIDE) [inline ebmlite/encoding.py:28]: Do internally-used constants need typing?

File: ebmlite/encoding.py [hunk 3]
```diff
@@ -218,7 +219,7 @@ def encodeBinary(val, length=None):
                          (len(val), length))
 
 
-def encodeString(val, length=None):
+def encodeString(val: ByteString, length: Optional[int] = None):
```
Comment (StokesMIDE) [inline ebmlite/encoding.py:222]: Should also include `str`... from the EBML spec, the subset-of-ASCII limitation is on the storage, and the data is still considered to be strings.

**Review by StokesMIDE — COMMENTED** [review]
Is the `mypy` and `.pyi` stuff still required if we're doing standard Python 3.5 typing?

File: ebmlite/encoding.py (hunk 2 above)
Comment (CFlaniganMide) [reply ebmlite/encoding.py:28]: I probably got a little over-zealous when I originally did this, honestly.

**Comment by StokesMIDE** [comment]
Annotation done in PR #127, closing this one.

**Pull request closed without merging**